lambda/src/main/scala/pricemigrationengine/handlers/CohortHandler.scala

```diff
@@ -32,33 +32,33 @@ trait CohortHandler extends zio.App with RequestStreamHandler {
     (for {
       cohortSpec <-
         ZIO
-          .effect(read[CohortSpec](input))
+          .attempt(read[CohortSpec](input))
           .mapBoth(e => InputFailure(s"Failed to parse json: $e"), Option(_))
-          .filterOrElse(_.exists(CohortSpec.isValid))(spec => ZIO.fail(InputFailure(s"Invalid cohort spec: $spec")))
+          .filterOrElseWith(_.exists(CohortSpec.isValid))(spec => ZIO.fail(InputFailure(s"Invalid cohort spec: $spec")))
       validSpec <- ZIO.fromOption(cohortSpec).orElseFail(InputFailure("No input"))
     } yield validSpec).tapBoth(
       e => Logging.error(e.toString),
       spec => Logging.info(s"Input: $spec")
     )
 
-  final def run(args: List[String]): ZIO[ZEnv, Nothing, ExitCode] =
+  override final def run: ZIO[ZEnv with ZIOAppArgs, Any, Any] =
```

Comment:
why has this been changed to override?

Reply:
It was always overriding the `run` method in the trait.  I just thought it might be better to make that explicit so that the compiler can help us out.